reactions/src/HALO.cpp

```diff
@@ -116,9 +132,35 @@ void HALO::scol_init(double vars[]) const{
 // arrays to store values
 double lgmass[loop_N],sigar[loop_N],scolar0[loop_N],scolar1[loop_N],scolar2[loop_N];
 
-// calculate sigma_8^2 and it's smoothing scale derivative @ R=8.
-double sig1 = Integrate<ExpSub>(bind(sigma_integrand, cref(P_l), 8., std::placeholders::_1), 1e-4, 50., 1e-3);
-double sig2 = Integrate<ExpSub>(bind(sigma8d_integrand, cref(P_l), 8., std::placeholders::_1), 1e-4, 50., 1e-3);
+// initialise linear growth spline for the theory of gravity/dark energy. See specialfunctions.cpp mu,gamma2,gamma3 functions
+int loop_Nk = 300;
+double kmin = 1e-5;
+double kmax = 100.;
+double kval_tab[loop_Nk],ling_tab[loop_Nk];
+
+
+for(int i = 0; i< loop_Nk; i++){
+  kval_tab[i] =  kmin * exp(i*log(kmax/kmin)/(loop_Nk-1.));
+  // numerical initialisation of linear growth in general theory of gravity/dark energy - see SpecialFunctions.cpp
+  iow.initn_lin(vars[0], kval_tab[i], vars[1],vars[2], vars[3], vars[4]);
+  ling_tab[i] = F1_nk/dnorm_spt;
+    }
+
+// spline linear growth factor
+ linear_growth = LinearSpline(loop_Nk,kval_tab,ling_tab);
+
+ // select what variance to use  (if input P(k) is LCDM at z=0, vars[6]=0, if P(k) if modified at z=z, vars[6]=1)
+ double sig1,sig2;
+
+  if (vars[6] < 0.5) {
```

Comment:
is this a bit dangerous perhaps? why 0.5 and not a very small number since the "truth" is 0 or 1?

Reply:
Agree. Made it a bool variable/flag (mgcamb) now so it's either true or false.